tiledb/sm/array_schema/dimension.cc

```diff
@@ -234,8 +235,8 @@ void Dimension::dump(FILE* out) const {
   if (out == nullptr)
     out = stdout;
   // Retrieve domain and tile extent strings
-  std::string domain_s = utils::parse::domain_str(domain_, type_);
-  std::string tile_extent_s =
+  const std::string& domain_s = utils::parse::domain_str(domain_, type_);
```

Comment:
The const-ref is extraneous because `domain_str` does not return a reference. Please remove it here and elsewhere in the patch where const-ref variables are assigned to a return-by-value function. All modern compilers will not copy the return value in this scenario anyway: https://en.wikipedia.org/wiki/Copy_elision

I'm all for const-correctness, so `const std::string` would be appropriate it here.

Reply:
Thanks for the reply, @joe-maley. I knew that copy elision would be done mostly in those cases but not guaranteed with C++11 standards. Only with C++17 standards, no need to use ref-variables, (because of that reason mostly they'd prefer use to auto here).

For those, maybe this [link](https://en.cppreference.com/w/cpp/language/copy_elision#Non-mandatory_elision_of_copy.2Fmove_.28since_C.2B.2B11.29_operations) would be helpful to understand copy elision deeply. However, it is always OK for me to following rules/suggestions from merge request :)

Reply:
That's right -- I was referring to NRVO which isn't in the standard but has been implemented in gcc and clang for a while.